mep/accounts/models.py

```diff
@@ -398,48 +398,46 @@ class DatePrecision(Flags):
 
 class DatePrecisionField(models.PositiveSmallIntegerField):
 
-    DATE_FORMATS = {
-        DatePrecision.year: '%Y',
-        DatePrecision.month: '%Y-%m',
-        DatePrecision.day: '%Y-%m-%d'
-    }
-
-    partial_date_re = re.compile(
-       r'^(?P<year>\d{4})?(?:-(?P<month>[01]\d))?(?:-(?P<day>[0-3]\d))?$'
-    )
-
     def __init__(self, *args, **kwargs):
-        if 'default' not in kwargs:
-            # default is full precision
-            # NOTE: needs to be cast as int or it breaks django migrations
-            kwargs['default'] = int(DatePrecision.all_flags)
-        super(DatePrecisionField, self).__init__(*args, **kwargs)
 
-    @classmethod
-    def parse_date(cls, value):
-        '''Parse a partial date string and return a :class:`datetime.date`
-        and precision value.'''
-        # partial date parsing adapted in part from django_partial_date
-        # https://github.com/ktowen/django_partial_date
-        match = cls.partial_date_re.match(value)
-        if match:
-            match_info = match.groupdict()
+        if 'default' not in kwargs:
+            # default is no precision
+            kwargs['default'] = None
```

Comment:
I think we probably don't need to specify this; now that null is allowed, I would expect `None` to be the default.  I think that means we don't need a custom `__init__` method at all anymore.

Reply:
The one thing we probably should add to the `DatePrecisionField` is a `to_python` method that casts the `int` to  a `DatePrecision` flag class.  I wondered about doing that on my feature branch but didn't get to it before you started working on it. 
https://docs.djangoproject.com/en/2.0/howto/custom-model-fields/#converting-values-to-python-objects